terraform/waf.tf

```diff
@@ -0,0 +1,122 @@
+﻿resource "aws_wafv2_regex_pattern_set" "ehrc_protected_request_address" {
+  name        = "ehrc-rotected-request-address"
+  description = "Regex of the endpoint used by ehrc. Of the form .../download?p=filename"
+  scope       = "REGIONAL"
+
+  regular_expression {
+    regex_string = "^/download(/)?[?]p=(.*)$"
+  }
+}
+
+resource "aws_wafv2_ip_set" "ehrc_whitelisted_ips" {
+  name               = "ehrc-whitelisted-ips"
+  description        = "EHRC whitelisted IPs. Only these IPs can access the protected endpoint of the form .../download?p=filename"
+  scope              = "REGIONAL"
+  ip_address_version = "IPV4"
+  addresses          = [for ip in split("\n", file("EHRCDownload-IP-Whitelist.txt")): trimspace(ip)]
+}
+
+resource "aws_wafv2_ip_set" "blacklisted_ips" {
+  name               = "blacklisted-ips"
+  description        = "Blacklisted IPs. These IPs cannot connect to the website."
+  scope              = "REGIONAL"
+  ip_address_version = "IPV4"
+  addresses          = [for ip in split("\n", file("GPG-IP-Denylist.txt")) : trimspace(ip)]
+}
```

Comment:
These files will be downloaded by the azure pipeline

Reply:
What's the process for updating these files? I know they haven't changed in a while, but I think now's a good time to make sure we're considering that so they can be updated easily in the future.

Reply:
The files are on keeper. I'm not aware of any particular way of updating them. I suppose one can download them from keeper, make the changes, and then upload them back on keeper and azure.

Reply:
Hmm ok. I think because of the time pressure and the fact these files don't change often we can leave it like this.
When it comes to documentation though, I think there should be a big "How to update the IP blacklist/whitelist" heading with detailed instructions on where the files can be found, how to replace them in Azure, and how to make sure those changes are deployed. Just something to remember when we get to that point 🙂